### PR TITLE
MTDSA-4761 Code coverage to pass level

### PR DESCRIPTION
**Pull request by narendravijayarao**

### CODE REVIEW

File: conf/logback-test.xml [hunk 1]
```diff
@@ -42,13 +42,13 @@
     </appender>
 
     <logger name="application" level="DEBUG"/>
-    <logger name="uk.gov" level="ERROR"/>
+    <logger name="uk.gov" level="DEBUG"/>
```
Comment (james-work-account) [inline conf/logback-test.xml]: Test logs already spit out much more than is necessary - why change to DEBUG?

File: app/uk/gov/hmrc/vatapi/filters/AgentSimulationFilter.scala [hunk 2]
```diff
@@ -30,7 +32,8 @@ class AgentSimulationFilter @Inject()(implicit val mat: Materializer, ec: Execut
     val method = rh.method
 
     rh.headers.get(GovTestScenarioHeader) match {
-      case Some("CLIENT_OR_AGENT_NOT_AUTHORISED") => ClientSubscriptionSimulation(f, rh, method)
+      case Some("CLIENT_OR_AGENT_NOT_AUTHORISED") =>
```
Comment (james-work-account) [inline app/uk/gov/hmrc/vatapi/filters/AgentSimulationFilter.scala]: implicit ExecutionContext can be removed from class @ Inject now

File: app/uk/gov/hmrc/vatapi/utils/ErrorHandler.scala [hunk 3]
```diff
@@ -16,57 +16,51 @@
...
-            case ex: NotImplementedException =>
-              NotImplemented(Json.toJson(ErrorNotImplemented))
-            case _ => result
-          }
-      }
-    }
-  }
+  override def onClientError(request: RequestHeader, statusCode: Int, message: String): Future[Result] = {
+
+    implicit val headerCarrier: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
```
Comment (james-work-account) [inline app/uk/gov/hmrc/vatapi/utils/ErrorHandler.scala]: unused - can be removed

File: test/uk/gov/hmrc/vatapi/UnitSpec.scala [hunk 4]
```diff
@@ -17,6 +17,7 @@
 package uk.gov.hmrc.vatapi
 
 import org.joda.time.{DateTime, DateTimeZone, LocalDate}
+import org.scalamock.scalatest.MockFactory
```
Comment (james-work-account) [inline test/uk/gov/hmrc/vatapi/UnitSpec.scala]: unused - can be removed

File: test/uk/gov/hmrc/vatapi/utils/ErrorHandlerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.audit.http.connector.AuditResult.Success
+import uk.gov.hmrc.play.audit.model.DataEvent
+import uk.gov.hmrc.play.bootstrap.config.HttpAuditEvent
+import uk.gov.hmrc.vatapi.UnitSpec
+import uk.gov.hmrc.vatapi.models.{ErrorBadRequest, ErrorCode, ErrorNotImplemented}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NoStackTrace
```
Comment (james-work-account) [inline test/uk/gov/hmrc/vatapi/utils/ErrorHandlerSpec.scala]: unused - can be removed

File: conf/logback-test.xml (hunk 1 above)
Comment (narendravijayarao) [reply conf/logback-test.xml]: Not sure what you are asking... I have changed from ERROR to DEBUG

**Pull request merged**